crates/sp-domains/src/fraud_proof.rs

```diff
@@ -232,7 +234,8 @@ impl<Number, Hash, DomainNumber, DomainHash>
         Self {
             domain_id,
             bad_receipt_hash,
-            consensus_block_hash,
+            consensus_block_incl_er,
+            consensus_block_incl_bundle,
```

Comment:
The fraud proof is targetting an invalid ER, `consensus_block_incl_bundle` should be `ER::consensus_block_hash` and why do we need `consensus_block_incl_er`?

Reply:
I think it is okay to use the best consensus hash to get the bad receipt, because the fraud proof is included in the next consensus block that builds on top of the best consensus block so the fraud proof should be verified against the best consensus block, if the bad receipt don't exist here which means the bad receipt is already pruned or is in other forks, in both cases the fraud proof should be considered as invalid.